sdcm/cluster_baremetal.py

```diff
@@ -17,9 +20,9 @@ class NodeIpsNotConfiguredError(Exception):
 class PhysicalMachineNode(cluster.BaseNode):
     log = LOGGER
 
-    def __init__(self, name, parent_cluster, public_ip, private_ip, credentials, base_logdir=None, node_prefix=None):  # pylint: disable=too-many-arguments
+    def __init__(self, name, parent_cluster: 'PhysicalMachineCluster', public_ip, private_ip, credentials, base_logdir=None, node_prefix=None):  # pylint: disable=too-many-arguments
         ssh_login_info = {'hostname': None,
-                          'user': credentials.name,
```

Comment:
We need to be extra careful in changing baremetal implementation, since the siren tests are using it, so test it with siren at some point before merging this

Reply:
credentials.name here is ssh keyfile comment - it doesn't seem it would work with pretty much any key we have.

Reply:
Can you change it to something like `getattr(self.parent_cluster, "ssh_username", credentials.name)` ?